cloudigrade/util/aws.py

```diff
@@ -63,18 +63,64 @@ def is_running(cls, code):
         return code == cls.running.value
 
 
-def extract_account_id_from_arn(arn):
+class AwsArn(object):
     """
-    Extract the AWS account ID from the given ARN.
+    Object representing an AWS ARN.
 
-    Args:
-        arn (str): A well-formed Amazon Resource Name.
+    See also:
+        https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
 
-    Returns:
-        Decimal: The AwsAccount ID found in the ARN.
+    General ARN formats:
+        arn:partition:service:region:account-id:resource
+        arn:partition:service:region:account-id:resourcetype/resource
+        arn:partition:service:region:account-id:resourcetype:resource
+
+    Example ARNs:
+        <!-- Elastic Beanstalk application version -->
+        arn:aws:elasticbeanstalk:us-east-1:123456789012:environment/My App/MyEnvironment
+
+        <!-- IAM user name -->
+        arn:aws:iam::123456789012:user/David
+
+        <!-- Amazon RDS instance used for tagging -->
+        arn:aws:rds:eu-west-1:123456789012:db:mysql-db
+
+        <!-- Object in an Amazon S3 bucket -->
+        arn:aws:s3:::my_corporate_bucket/exampleobject.png
 
     """
-    return decimal.Decimal(arn.split(':')[4])
+
+    # pylint: disable=line-too-long
```

Comment:
Please don't disable pylint. Breaking up the mile-long regex into a few lines would also improve its readability.

Reply:
In addition, this project doesn't include pylint as part of its CI, so this doesn't accomplish much.